shared/cert.go

```diff
@@ -259,15 +259,26 @@ func GenCert(certf string, keyf string, certtype bool, addHosts bool) error {
 	if err != nil {
 		return fmt.Errorf("Failed to open %s for writing: %w", certf, err)
 	}
-	certOut.Write(certBytes)
-	certOut.Close()
+
+	defer certOut.Close()
```

Comment:
As this is a potential write it will need its error handled.

Reply:
@tomponline I'm going through all of the `file.Close()` type defers and errors now and in general I am allowing the defer only when the file is only opened for reading. In those cases I am replacing with `defer func() { _ = file.Close() }()`. When a file is opened for writing I am still deferring and ignoring the deferred error, but also closing manually and checking the error.

Reply:
Very good.